consumers/campaigner/main.go

```diff
@@ -67,127 +91,121 @@ func (h *MessageHandler) HandleMessage(m *nsq.Message) error {
+				logrus.WithError(err).
+					WithField("subscriber", s).
+					Error("Unable to get subscriber metadata.")
 				continue
 			}
+			//  todo merge def template data with sub metadata
 
-			// prepare message for publishing to the queue
-			input := &ses.SendBulkTemplatedEmailInput{
-				Source:              aws.String(msg.Source),
-				Template:            aws.String(msg.Campaign.Template.Name),
-				Destinations:        dest,
-				DefaultTemplateData: aws.String(string(defaultData)),
-				DefaultTags: []*ses.MessageTag{
-					&ses.MessageTag{
-						Name:  aws.String("campaign_id"),
-						Value: aws.String(strconv.FormatInt(msg.Campaign.ID, 10)),
-					},
-					&ses.MessageTag{
-						Name:  aws.String("user_id"),
-						Value: aws.String(msg.UserUUID),
-					},
-				},
-			}
+			// fill template buffer with rendered template.
+			var (
+				htmlBuf bytes.Buffer
+				subBuf  bytes.Buffer
+				textBuf bytes.Buffer
+			)
 
-			if msg.ConfigurationSetExists {
-				input.ConfigurationSetName = aws.String(emails.ConfigurationSetName)
+			err = html.FRender(&htmlBuf, m)
+			if err != nil {
+				logrus.WithError(err).Error("unable to render html_part")
+			}
+			err = txt.FRender(&subBuf, m)
+			if err != nil {
+				logrus.WithError(err).
+					WithField("subject_part", template.SubjectPart).
+					Error("unable to render text_part")
+			}
+			err = sub.FRender(&textBuf, m)
+			if err != nil {
+				logrus.WithError(err).
+					WithField("subject_part", template.SubjectPart).
+					Error("unable to render subject_part")
 			}
 
-			uuid := uuid.New()
-
-			bulkMsg := entities.BulkSendMessage{
-				UUID:       uuid.String(),
-				Input:      input,
-				CampaignID: msg.Campaign.ID,
-				UserID:     msg.UserID,
-				SesKeys:    &msg.SesKeys,
+			sender := entities.SendEmailTopicParams{
+				UUID:         uuid.New().String(),
+				SubscriberID: s.ID,
+				CampaignID:   campaign.ID,
+				SesKeys:      msg.SesKeys,
+				HTMLPart:     htmlBuf.Bytes(),
+				SubjectPart:  subBuf.Bytes(),
+				TextPart:     textBuf.Bytes(),
+				UserUUID:     msg.UserUUID,
+				UserID:       msg.UserID,
 			}
-			msg, err := json.Marshal(bulkMsg)
 
+			senderBytes, err := json.Marshal(sender)
 			if err != nil {
 				logrus.WithError(err).Error("Unable to marshal bulk message input.")
 				continue
 			}
 
 			// publish the message to the queue
-			err = h.p.Publish(entities.SendBulkTopic, msg)
+			err = h.p.Publish(entities.SenderTopic, senderBytes)
 			if err != nil {
 				logrus.WithError(err).Error("Unable to publish message to send bulk topic.")
+				continue
 			}
 		}
 
+		// set  vars for next batches
 		lastSub := subs[len(subs)-1]
 		nextID = lastSub.ID
 		timestamp = lastSub.CreatedAt
-	}
 
-	c := msg.Campaign
-	c.UserID = msg.UserID
-	c.Status = entities.StatusSent
-	c.CompletedAt.SetValid(time.Now().UTC())
+		// exit the loop if next batch is smaller then 1k -> it's the last batch of 1k.
+		if len(subs) < 1000 {
+			continue
```

Comment:
Did you mean to `break` here? Since continue does nothing because this is at the end of the loop.